packages/@remirror/extension-annotation/src/annotation-extension.ts

```diff
@@ -0,0 +1,197 @@
+            return newDecorationSet.remove(found);
+          }
+
+          if (actionType === ActionType.SET_ANNOTATIONS) {
+            const setAction = action as SetAnnotationsAction<A>;
+            const decos = setAction.annotations.map((a) => {
+              // Ignore field "text". Prosemirror content is source of truth.
+              const { from, to, text, ...annotationData } = a;
+              return toDecoration({
+                from,
+                to,
+                annotationData,
+                annotationClassName: this.options.annotationClassName,
+              });
+            });
+            return DecorationSet.create(tr.doc, decos);
+          }
+
+          return newDecorationSet;
+        },
+      },
+      props: {
+        decorations(state: EditorState) {
+          return this.getState(state);
+        },
+      },
+    };
+  }
+
+  createCommands() {
+    return {
+      /**
+       * Adds an annotation spanning the currently selected content
+       */
+      addAnnotation: (annotationData: AnnotationData<A>): CommandFunction => ({
+        state,
+        dispatch,
+      }) => {
+        const sel = state.selection;
+
+        if (sel.empty) {
+          return false;
+        }
+
+        if (dispatch) {
+          const action: AddAnnotationAction<A> = {
+            type: ActionType.ADD_ANNOTATION,
+            from: sel.from,
+            to: sel.to,
+            annotationData,
+          };
+          dispatch(state.tr.setMeta(AnnotationExtension.name, action));
+        }
+
+        return true;
+      },
+
+      /**
+       * Removes a list of annotations
+       */
+      removeAnnotations: (ids: string[]): CommandFunction => ({ state, dispatch }) => {
+        if (dispatch) {
+          const action: RemoveAnnotationsAction = {
+            type: ActionType.REMOVE_ANNOTATIONS,
+            annotationIds: ids,
+          };
+          dispatch(state.tr.setMeta(AnnotationExtension.name, action));
+        }
+
+        return true;
+      },
+
+      /**
+       * Sets the annotation. Use this to initialize the extension based on
+       * loaded data
+       */
+      setAnnotations: (annotations: A[]): CommandFunction => ({ state, dispatch }) => {
+        if (dispatch) {
+          const action: SetAnnotationsAction<A> = {
+            type: ActionType.SET_ANNOTATIONS,
+            annotations,
+          };
+          dispatch(state.tr.setMeta(AnnotationExtension.name, action));
+        }
+
+        return true;
+      },
+    };
+  }
+
+  /**
+   * @returns all annotations in the editor
+   */
+  getAnnotations(state: EditorState): A[] {
+    const decorations: DecorationSet = this.getPluginState();
+    return decorations.find().map((decoration) =>
+      toAnnotation({
+        state,
+        decoration,
+      }),
+    );
+  }
+
+  /**
+   * @returns all annotations at a specific position in the editor
+   */
+  getAnnotationsAt(state: EditorState, pos: number): A[] {
```

Comment:
Add this to the `createHelpers` method as well so that users can do this.

```ts
const { helpers } = useRemirror();

const annotations = getAnnotationsAt(5);
```

Also instead of requiring the users pass in the state you can use the `this.store.getState()` in this method.

```suggestion
  getAnnotationsAt(pos: number): A[] {
```